### PR TITLE
feat: use k8s cluster config when buildx used in k8s cluster

### PR DESCRIPTION
**Pull request by morlay**
grant permission to `service account` of pod.

### CODE REVIEW

**Review by AkihiroSuda — COMMENTED** [review]
Looks good,thanks, but please sign with full real name?

File: driver/manager.go [hunk 1]
```diff
@@ -22,11 +24,30 @@ type BuildkitConfig struct {
...
+func (k KubeClientConfigInCluster) ClientConfig() (*rest.Config, error) {
+	return rest.InClusterConfig()
+}
+
+func (k KubeClientConfigInCluster) Namespace() (string, bool, error) {
+	namespace, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+	if err != nil {
+		return "", false, nil
+	}
+	return string(namespace), true, nil
```
Comment (AkihiroSuda) [inline driver/manager.go:43]: strings.TrimSpace

File: driver/manager.go [hunk 2]
```diff
@@ -22,11 +24,30 @@ type BuildkitConfig struct {
...
+type KubeClientConfigInCluster struct{}
+
+func (k KubeClientConfigInCluster) ClientConfig() (*rest.Config, error) {
+	return rest.InClusterConfig()
+}
+
+func (k KubeClientConfigInCluster) Namespace() (string, bool, error) {
+	namespace, err := ioutil.ReadFile("/var/run/secrets/kubernetes.io/serviceaccount/namespace")
+	if err != nil {
+		return "", false, nil
```
Comment (AkihiroSuda) [inline driver/manager.go:41]: Make sure to return err

**Review by AkihiroSuda — CHANGES_REQUESTED** [review]

**Comment by morlay** [comment]
Github not allow me to change it.

image

 i used this name since i join github. it could represent myself

**Review by AkihiroSuda — APPROVED** [review]

**Pull request merged**